woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1727,3 +1727,56 @@ def test_sets_koalas_option_on_init(sample_df_koalas):
         ks.set_option('compute.ops_on_diff_frames', False)
         sample_df_koalas.ww.init()
         assert ks.get_option('compute.ops_on_diff_frames') is True
+
+
+def test_maintain_column_order_on_type_changes(sample_df):
```

Comment:
Do we expect that any of these operations would change the order of either the schema columns or the dataframe columns? Doesn't seem like it to me, so I think this test would pass even without the changes to the accessor code. Having said that, I'm fine with leaving this as an additional validation check to make sure we don't break it in the future.